src/app/cli/src/init/test_block_command_verification.ml

```diff
@@ -0,0 +1,123 @@
+  printf
+    "If validation fails you may need to compile the test with a different \
+     profile\n" ;
+  printf "Reading block from %s\n" large_precomputed_json_file ;
+  let json =
+    Yojson.Safe.from_string (In_channel.read_all large_precomputed_json_file)
+  in
+  let precomputed =
+    match Mina_block.Precomputed.of_yojson json with
+    | Ok json ->
+        json
+    | Error err ->
+        failwith err
+  in
+  let logger = Logger.create () in
+  let commit_id = "<not-needed-for-test>" in
+  Logger.Consumer_registry.register ~id:Logger.Logger_id.mina
+    ~processor:Internal_tracing.For_logger.processor ~commit_id
+    ~transport:
+      (Logger_file_system.dumb_logrotate ~directory:log_directory
+         ~log_filename:"internal-tracing.log"
+         ~max_size:(1024 * 1024 * 10)
+         ~num_rotate:50 )
+    () ;
+  let%bind () =
+    Internal_tracing.toggle ~commit_id ~force:true ~logger `Enabled
+  in
+  let proof_cache_db = Proof_cache_tag.For_tests.create_db () in
+  let commands =
+    precomputed.staged_ledger_diff
+    |> Staged_ledger_diff.write_all_proofs_to_disk ~proof_cache_db
+    |> Staged_ledger_diff.commands
+  in
+  let find_vk _frozen_ledger_hash account_id =
+    (let open Option.Let_syntax in
+    let%bind account =
+      List.find_map precomputed.accounts_accessed ~f:(fun (_idx, acnt) ->
+          let acnt_id = Mina_base.Account.identifier acnt in
+          if Mina_base.Account_id.equal acnt_id account_id then Some acnt
+          else None )
+    in
+    let%bind zkapp = account.zkapp in
+    zkapp.verification_key)
+    |> Option.value_map
+         ~default:(Or_error.error_string "Zkapp verification key not found")
+         ~f:Or_error.return
+  in
+  let verifiable_commands =
+    List.map commands ~f:(fun command ->
+        let verifiable_command =
+          Mina_base.User_command.to_verifiable ~failed:false ~find_vk
+            command.data
+          |> Or_error.ok_exn
+        in
+        { Mina_base.With_status.data = verifiable_command
+        ; status = command.status
+        } )
+  in
+  let proof_level = Genesis_constants.Proof_level.Full in
+  let constraint_constants = Genesis_constants.Compiled.constraint_constants in
+  let module T = Transaction_snark.Make (struct
+    let constraint_constants = constraint_constants
+
+    let proof_level = proof_level
+
+    let signature_kind = signature_kind
+  end) in
+  let module B = Blockchain_snark.Blockchain_snark_state.Make (struct
+    let tag = T.tag
+
+    let constraint_constants = constraint_constants
+
+    let proof_level = proof_level
+  end) in
+  let%bind blockchain_verification_key = Lazy.force B.Proof.verification_key in
+  let%bind transaction_verification_key = Lazy.force T.verification_key in
+  let%bind verifier =
+    (* The Proof_level.Full dummy verifier uses the same
+       Pickles.Side_loaded.verify function that the Prod verifier does, which is
+       what we want to test here. *)
+    Verifier.Dummy.create ~logger
```

Comment:
I suggest make this configurable. I remember there's some behavioral difference between prod verifier and dummy verifier. But yes, prod verifier needed some parallel lib set up

Reply:
I remember there being more than just `Parallel.init_master ()` required to use the prod verifier interface - I think the actual verifier child process needs to be spawned too.

George thought it would be simpler for this benchmark not to spawn the full verifier. There is a common implementation for the prod/dummy verifier now, so I'm able to call `Verifier.Dummy.create` with the `Full` proof level and this does work. It just invokes the verification code directly without needing to talk to a separate verifier process.